packages/git/src/index.ts

```diff
@@ -182,6 +182,22 @@ export async function isRepoShallow({ cwd }: { cwd: string }) {
 export async function deepenCloneBy({ by, cwd }: { by: number; cwd: string }) {
   await spawn("git", ["fetch", `--deepen=${by}`], { cwd });
 }
+async function getRepoRoot({ cwd }: { cwd: string }) {
+  const { stdout, code } = await spawn(
+    "git",
+    ["rev-parse", "--show-toplevel"],
+    { cwd }
+  );
+
+  if (code !== 0) {
+    return cwd;
+  }
```

Comment:
according to the docs:
> Show the (by default, absolute) path of the top-level directory of the working tree. If there is no working tree, report an error.

So I think that we should throw an error here:
```suggestion
  if (code !== 0) {
    throw new Error(gitCmd.stderr.toString());
  }
```

Where for us this actually shouldn't ever throw, but this function is quite generic - so if we ever expose it then somebody could call it with `cwd` outside of a git repo.

I think that it's better to throw here to raise potential issues early, rather than to swallow the error and lead to undetermined behavior.

Reply:
Sounds reasonable